service/tracker/consensus_test.go

```diff
@@ -0,0 +1,908 @@
+// Copyright 2021 Optakt Labs OÜ
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+package tracker_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/flow-go/engine/execution/computation/computer/uploader"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/storage/badger/operation"
+
+	"github.com/optakt/flow-dps/service/tracker"
+	"github.com/optakt/flow-dps/testing/helpers"
+	"github.com/optakt/flow-dps/testing/mocks"
+)
+
+func TestConsensus_Root(t *testing.T) {
+	header := mocks.GenericHeader
+
+	t.Run("nominal case", func(t *testing.T) {
+		t.Parallel()
+
+		db := helpers.InMemoryDB(t)
+		defer db.Close()
+
+		require.NoError(t, db.Update(operation.InsertRootHeight(header.Height)))
+
+		cons := tracker.BaselineConsensus(t, tracker.WithDB(db))
+
+		got, err := cons.Root()
+
+		require.NoError(t, err)
+		assert.Equal(t, header.Height, got)
+	})
+
+	t.Run("handles missing root height", func(t *testing.T) {
+		t.Parallel()
+
+		db := helpers.InMemoryDB(t)
+		defer db.Close()
+
+		// Root height missing from database.
+		//require.NoError(t, db.Update(operation.InsertRootHeight(header.Height)))
```

Comment:
Should we have these pieces of codes committed but commented out? I feel like the comment above says enough.

Reply:
I thought it makes it clearer exactly what is omitted but if you guys think it's clear enough with just the comment, I'm fine with removing them :)